pygmt/src/meca.py

```diff
@@ -211,6 +213,12 @@ def meca(
         **+s**\ *size* to set the diameter of the circle [Default is
         no circle]. Use **+p**\ *pen* to set the line pen attributes [Default
         is 0.25p].
+    compressionfill : str
+        Select color or pattern for filling the compressional quadrants
+        [Default is black].
+    extensionfill : str
+        Select color or pattern for filling the extensive quadrants
```

Comment:
```suggestion
        Set fill shade, color, or pattern for the extensive quadrants
```
To be consistent with e.g. the formulation in Figure.wiggle.

Reply:
We should also add at least one unit test to check if the fill works as expected.

Reply:
I recognized already yesterday that the docstrings for `fill` are not consistent across the documentation.
So far I orientated myself on the one of `pygmt.Figure.plot`, as I was not sure about the meaning of the term "shade" and whether it is available for filling quadrants of beach balls (I had not time to check it until now - I apologize for this :slightly_frowning_face:).

However, I agree, that there should be a consistent formulation across the different plotting methods (with adjustments in case the available functionality is partly different), e.g.,:

- https://www.pygmt.org/latest/api/generated/pygmt.Figure.plot.html#pygmt.Figure.plot
> fill ([str](https://docs.python.org/3/library/stdtypes.html#str)) – Select color or pattern for filling of symbols or polygons [Default is no fill]. fill can be a 1-D array, but it is only valid if using x/y and cmap=True is also required.

- https://www.pygmt.org/latest/api/generated/pygmt.Figure.plot3d.html#pygmt.Figure.plot3d
> fill ([str](https://docs.python.org/3/library/stdtypes.html#str)) – Select color or pattern for filling of symbols or polygons [Default is no fill]. fill can be a 1-D array, but it is only valid if using x/y and cmap=True is also required.

- https://www.pygmt.org/latest/api/generated/pygmt.Figure.velo.html
> uncertaintyfill ([str](https://docs.python.org/3/library/stdtypes.html#str)) – Sets the color or shade used for filling uncertainty wedges (spec="w") or velocity error ellipses (spec="e" or spec="r"). If uncertaintyfill is not specified, the uncertainty regions will be transparent. Note: Using cmap and zvalue="+e" will update the uncertainty fill color based on the selected measure in zvalue [magnitude error]. More details at https://docs.generic-mapping-tools.org/latest/cookbook/features.html#gfill-attrib.

- https://www.pygmt.org/latest/api/generated/pygmt.Figure.wiggle.html#pygmt.Figure.wiggle
> fillpositive ([str](https://docs.python.org/3/library/stdtypes.html#str)) – Set fill shade, color, or pattern for positive wiggles [Default is no fill].
> fillnegative ([str](https://docs.python.org/3/library/stdtypes.html#str)) – Set fill shade, color, or pattern for negative wiggles [Default is no fill].

- https://www.pygmt.org/latest/api/generated/pygmt.Figure.solar.html#pygmt.Figure.solar
> fill ([str](https://docs.python.org/3/library/stdtypes.html#str)) – Color or pattern for filling of terminators.

- https://www.pygmt.org/latest/api/generated/pygmt.Figure.histogram.html#pygmt.Figure.histogram
> fill ([str](https://docs.python.org/3/library/stdtypes.html#str)) – Select color or pattern for filling of symbols or polygons [Default is no fill].

- https://www.pygmt.org/latest/api/generated/pygmt.Figure.rose.html#pygmt.Figure.rose
> fill ([str](https://docs.python.org/3/library/stdtypes.html#str)) – Selects shade, color or pattern for filling the sectors [Default is no fill].

Reply:
Based on the comparison of the docstrings for ``fill``, we have to decide regarding consistency:
- **verb**: "set" or "select"
- **grammar**: third person "s" or not (this is not consistent across parameters as well as functions/methods in the docs)
- **formulation structure**: "fill" or "for filling"
- **content**: mention "shade" explicitly or not (I guess it simply refers to gray shade i.e. gray color (?))

@GenericMappingTools/pygmt-maintainers should we go with the version of @michaelgrund's code suggestion above, or what are the preferences here? In case this change is welcome, I can update the (other) docstrings for ``fill`` in a separate PR in the next days.

P.S.: @michaelgrund I apologize for my probably a bit harsh comment above!

Reply:
> as I was not sure about the meaning of the term "shade".

I also have no idea about "shade". Based on my understanding about this page (https://www.w3schools.com/colors/colors_shades.asp), I think "shades" means colors (not only gray) with different lights, so it's already covered by "colors".

So I would prefer not to mention "shades".

"Set fill color or pattern" or "Set color or pattern filling" (preferred) both sound good to me.

Reply:
> > as I was not sure about the meaning of the term "shade".
> 
> I also have no idea about "shade". Based on my understanding about this page (https://www.w3schools.com/colors/colors_shades.asp), I think "shades" means colors (not only gray) with different lights, so it's already covered by "colors".

Yep, something in this direction I was also thinking. But how do I change or vary the "light of the color" via ``fill``?
The upstream GMT documentation at https://docs.generic-mapping-tools.org/6.4/cookbook/features.html#specifying-area-fill-attributes makes a difference between gray shade and RGB/HSV/CMYK color:
> In the first case we may specify a gray shade (0–255), RGB color (r/g/b all in the 0–255 range or in hexadecimal #rrggbb), HSV color (hue-saturation-value in the 0–360, 0–1, 0–1 range), CMYK color (cyan/magenta/yellow/black, each ranging from 0–100%), or a valid color name; in that respect it is similar to specifying the pen color settings (see pen color discussion under Section [Specifying pen attributes](https://docs.generic-mapping-tools.org/6.4/cookbook/features.html#specifying-pen-attributes)).

Reply:
> But how do I change or vary the "light of the color" via `fill`?

Specifying a gray shade is as simple as giving a number from 0 to 255. For other colors, there is no easy to specify their shade. If you want a blue color with a different light, then you need to use other tools to get the name or RGB values of the color, and pass it to GMT.

xref: 

1. https://en.wikipedia.org/wiki/List_of_colors_by_shade. 
2. https://mdigi.tools/color-shades/#0000ff

Reply:
Thanks! That is similar to my understanding.
I feel that mention "shade" explicitly can cause some confusions, as shade is only available for gray. Thus, I also prefer not to mention "shade" in the docs and state only "color" and "pattern".

Reply:
I started updating the (other) docstrings for `fill` in PR #2351.